common/src/retry.rs

```diff
@@ -171,21 +404,34 @@ macro_rules! retry_async {
         #[allow(unused)]
         use $crate::retry::RetryableError;
         let mut attempts = 0;
+        let time_spent = $crate::time::Instant::now();
         let span = tracing::trace_span!("retry");
         loop {
             let span = span.clone();
             #[allow(clippy::redundant_closure_call)]
+            $retry.cooldown().await;
             let res = $code.instrument(span).await;
             match res {
                 Ok(v) => break Ok(v),
                 Err(e) => {
+                    $retry.last_err(&e);
+                    if (&e).needs_cooldown() {
+                        tracing::warn!("Hit {}, cooling down", e);
```

Comment:
If an app thinks they might be getting rate limited, the error we could look for in the logs would be "Hit {Error}, cooling down"? Is that right? would the `ResourceExhausted` error bubble up here, and be logged, or would the cooling down warnings be the main hint?

Reply:
that's correct, we would see ResourceExhausted bubbled from the GRPC call, the grpc code/message, and 'cooling down'